kademlia/node.py

```diff
@@ -94,8 +94,9 @@ def push(self, nodes):
             nodes = [nodes]
 
         for node in nodes:
-            distance = self.node.distanceTo(node)
-            heapq.heappush(self.heap, (distance, node))
+	    if node.id not in [n.id for n in self]:
```

Comment:
I think this might be a good opportunity to add a ` __contains__` method, so then you could do:

```python
if node not in self:
     ...
```